Replace auto_ptr (deprecated/removed in C++11/17) with unique_ptr to build with
C++17 as required by abseil-cpp (through libphonenumber).

Index: src/addressbook/libebook-contacts/e-phone-number-private.cpp
--- src/addressbook/libebook-contacts/e-phone-number-private.cpp.orig
+++ src/addressbook/libebook-contacts/e-phone-number-private.cpp
@@ -183,7 +183,7 @@ _e_phone_number_cxx_from_string (const gchar *phone_nu
 	g_return_val_if_fail (NULL != phone_number, NULL);
 
 	const std::string valid_region = _e_phone_number_cxx_make_region_code (region_code);
-	std::auto_ptr<EPhoneNumber> parsed_number(new EPhoneNumber);
+	std::unique_ptr<EPhoneNumber> parsed_number(new EPhoneNumber);
 
 	if (!_e_phone_number_cxx_parse (
 		phone_number, valid_region, &parsed_number->priv, error))
